include/lager.hrl

```diff
@@ -20,6 +20,48 @@
 -define(DEFAULT_SINK, lager_event).
 -define(ERROR_LOGGER_SINK, error_logger_lager_event).
 
+-define(METADATA(Extras), [{severity, info},
+                           {pid, self()},
+                           {node, node()},
+                           {module, ?MODULE},
+                           {function, ?FUNCTION_NAME},
+                           {function_arity, ?FUNCTION_ARITY},
+                           {file, ?FILE},
+                           {line, ?LINE} | Extras]).
+
+-define(log(Level, Format, Args, Safety),
+        lager:dispatch_log(?DEFAULT_SINK, Level, ?METADATA([]), Format, Args,
+                           ?DEFAULT_TRUNCATION, Safety)).
+-define(log(Level, Metadata, Format, Args, Safety),
+        lager:dispatch_log(?DEFAULT_SINK, Level, ?METADATA(Metadata), Format, Args,
+                           ?DEFAULT_TRUNCATION, Safety)).
+
+-define(debug(Format, Args), ?log(debug, Format, Args, safe)).
```

Comment:
Tiny bit concerned that lager's `?debug` and other log level macros might get caught up in a consuming project's macro definitions.

Reply:
True, `?lager_debug` would be fine too.